bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/connector/EpsonProjectorTcpConnector.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.epsonprojector.internal.connector;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.util.Date;
+
+import org.apache.commons.io.IOUtils;
+import org.openhab.binding.epsonprojector.internal.EpsonProjectorException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Connector for TCP communication.
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+public class EpsonProjectorTcpConnector implements EpsonProjectorConnector {
+
+    private static final Logger logger = LoggerFactory.getLogger(EpsonProjectorTcpConnector.class);
+
+    String ip = null;
+    int port = 10000;
+    Socket socket = null;
+    InputStream in = null;
+    OutputStream out = null;
+
+    public EpsonProjectorTcpConnector(String ip, int port) {
+        this.ip = ip;
+        this.port = port;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void connect() throws EpsonProjectorException {
+        logger.debug("Open connection to address'{}:{}'", ip, port);
+
+        try {
+            socket = new Socket(ip, port);
+            in = socket.getInputStream();
+            out = socket.getOutputStream();
+        } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception